File: C++/Program/Vehicle.cpp
/*
    Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan Praktikum 4 
    dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
    kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

// Memasukan library yang digunakan
#include <iostream>
#include <string>

// Deklarasi namespace
using namespace std;

/*
    Kelas Vehicle yang digunakan untuk merepresentasikan data Vehicle, Vehicle merupakan kelas induk dari kelas Car dan Motorcycle
*/

class Vehicle {
    private:
        string platNomor;   // Atribut platNomor untuk menyimpan nomor plat kendaraan
        string merk;        // Atribut merk untuk menyimpan merk kendaraan
        int tahunProduksi;  // Atribut tahunProduksi untuk menyimpan tahun produksi kendaraan
        string warna;       // Atribut warna untuk menyimpan warna kendaraan

    public:
        // Constructor
        Vehicle(){

        }

        // Constructor : Overloading
        Vehicle(string platNomor){
            this->platNomor = platNomor;
            merk = "";
            tahunProduksi = 0;
            warna = "";
        }

        // Constructor : Overloading
        Vehicle(string platNomor, string merk){
            this->platNomor = platNomor;
            this->merk = merk;
            tahunProduksi = 0;
            warna = "";
        }

        // Constructor : Overloading
        Vehicle(string platNomor, string merk, int tahunProduksi){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            warna = "";
        }

        // Constructor : Overloading
        Vehicle(string platNomor, string merk, int tahunProduksi, string warna){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            this->warna = warna;
        }

        // Method setter untuk set nilai setiap atribut pada kelas Vehicle
        void setPlatNomor(string platNomor){
            this->platNomor = platNomor;
        }

        void setMerk(string merk){
            this->merk = merk;
        }

        void setTahunProduksi(int tahunProduksi){
            this->tahunProduksi = tahunProduksi;
        }

        void setWarna(string warna){
            this->warna = warna;
        }

        // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
        string getPlatNomor(){
            return platNomor;
        }

        string getMerk(){
            return merk;
        }

        int getTahunProduksi(){
            return tahunProduksi;
        }

        string getWarna(){
            return warna;
        }

        // Destructor
        ~Vehicle(){

        }
};

File: PYTHON/Program/Vehicle.py
# Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan Parktikum 4 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
# kecurangan seperti yang telah dispesifikasikan. Aamiin.

class Vehicle:
    
    __platNomor = ""        # variabel privat untuk Plat Nomor
    __merk = ""             # variabel privat untuk Merk Kendaraan
    __tahunProduksi = 0     # variabel privat untuk Tahun Produksi
    __warna = ""            # variabel privat untuk Warna Kendaraan

    # konstruktor dengan parameter
    def __init__(self, platNomor, merk, tahunProduksi, warna):
        self.__platNomor = platNomor    # inisialisasi atribut Plat Nomor
        self.__merk = merk              # inisialisasi atribut Merk Kendaraan
        self.__tahunProduksi = tahunProduksi    # inisialisasi atribut Tahun Produksi
        self.__warna = warna            # inisialisasi atribut Warna Kendaraan

    # metode untuk mengatur Plat Nomor
    def set_platNomor(self, platNomor):
        self.__platNomor = platNomor

    # metode untuk mendapatkan Plat Nomor
    def get_platNomor(self):
        return self.__platNomor
    
    # metode untuk mengatur Merk Kendaraan
    def set_merk(self, merk):
        self.__merk = merk

    # metode untuk mendapatkan Merk Kendaraan
    def get_merk(self):
        return self.__merk
    
    # metode untuk mengatur Tahun Produksi
    def set_tahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi

    # metode untuk mendapatkan Tahun Produksi
    def get_tahunProduksi(self):
        return self.__tahunProduksi
    
    # metode untuk mengatur Warna Kendaraan
    def set_warna(self, warna):
        self.__warna = warna

    # metode untuk mendapatkan Warna Kendaraan
    def get_warna(self):
        return self.__warna
    
    # metode untuk mencetak informasi kendaraan
    def infoVehicle(self):
        print("Plat Nomor       :", self.__platNomor)
        print("Merk             :", self.__merk)
        print("Tahun Produksi   :", self.__tahunProduksi)
        print("Warna            :", self.__warna)